cclib/parser_properties/atombasis.py

```diff
@@ -0,0 +1,62 @@
+from cclib.parser_properties import utils
+from cclib.parser_properties.base_parser import base_parser
+
+import numpy as np
+
+
+class atombasis(base_parser):
+    """
+    Docstring? Units?
+    """
+
+    known_codes = ["gaussian"]
+
+    @staticmethod
+    def gaussian(file_handler, ccdata) -> list | None:
```

Comment:
This is too much to go over in a code review but this reminds me of https://craftinginterpreters.com/representing-code.html#the-expression-problem: is it better for us to have properties as classes/programs as methods, or programs as classes/properties as methods?  This implements the former, since the latter has been a mess, and I think doing anything fancier (visitor) is not user-facing.

My one concern here is that this method is public.  Will it be able to be made private so a user never has to specialize their code for their specific program?  (Visibility _is_ a breaking API change.)